app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentUiRenderer.kt

```diff
@@ -29,11 +29,16 @@ class ScheduleAppointmentUiRenderer(
       }
     }
 
-    when (model.requestCompletedStatus) {
-      TeleconsultStatus.StillWaiting -> ui.showNextButton()
-      TeleconsultStatus.Yes -> ui.showDoneButton()
-      TeleconsultStatus.No -> ui.showDoneButton()
-      else -> ui.showDoneButton()
+    if (model.hasTeleconsultRecord) {
+      when (model.requestCompletedStatus) {
+        TeleconsultStatus.StillWaiting -> ui.showNextButton()
+        TeleconsultStatus.Yes -> ui.showDoneButton()
+        TeleconsultStatus.No -> ui.showDoneButton()
+        null -> ui.showNextButton()
+        is TeleconsultStatus.Unknown -> ui.showDoneButton()
+      }
```

Comment:
Let's extract this out into a separate method to fix the code climate issue of nested conditionals.